src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -282,7 +284,12 @@ public static TestResultSummary parseAndSummarize(@Nonnull JUnitTask task, Pipel
             }
 
             if (!task.isSkipPublishingChecks()) {
-                new JUnitChecksPublisher(build, task.getChecksName(), result, summary).publishChecks(listener);
+                // If we haven't been provided with a checks name, and we have pipeline test details, set the checks name
+                // to be a ' / '-joined string of the enclosing blocks names. JUnitChecksPublisher will handle defaults if
+                // checksName ends up being empty or null
+                String checksName = task.getChecksName() != null || pipelineTestDetails == null ? task.getChecksName() :
+                      StringUtils.join(new ReverseListIterator(pipelineTestDetails.getEnclosingBlockNames()), " / ");
```

Comment:
What if `enclosingBlockNames` is empty? (Possible, in fact common, in Scripted syntax.)

Reply:
That should result in an empty string being passed in, which will be fixed in the `JUnitChecksPublisher` ctor. Perhaps it would make more sense to stick the default here instead. Either way, this scenario should now be tested.